Timber Throwable shortcut

vanniktech opened the pull request:
Proposal for #145 

If you're okay with the API I'll start adding tests

JakeWharton commented:
Looks good

vanniktech commented on timber/src/test/java/timber/log/TimberTest.java:
```diff
@@ -384,6 +457,17 @@ private static void assertExceptionLogged(String message, String exceptionClassn
     assertThat(log.throwable).isNull();
   }
 
+  private static void assertExceptionLogged(int logType, String exceptionClassname, String tag) {
```
Dunno if we should combine this method with the one above or not.

JakeWharton replied:
Just have it call this method with a `null` tag and then change the tag assertion to
```java
assertThat(log.tag).isEqualTo(tag != null ? tag : "TimberTest");
```

vanniktech commented:
Added some tests. Seems like this is backwards compatible so this might not need a `5.0` version bump.

JakeWharton commented on timber/src/test/java/timber/log/TimberTest.java:
```diff
@@ -209,6 +210,78 @@
     assertExceptionLogged("OMFG!", "java.lang.NullPointerException");
   }
 
+  @Test public void shortcutExceptionVerbose() {
```
I would prefer to see all log levels put in a single test for each functionality. This way there's only two new tests. Also let's call these "exception only" instead of "shortcut exception".

JakeWharton commented:
Still looks good, just some small nits.

vanniktech commented:
Updated it now took me longer since I was on vacation. I merged the tests together and adjusted `assertExceptionLogged` though it feels clunky now given that it has a lot of parameters.

Let me know whether I should squash manually or you let Github do it. I find it easier to review it commit by commit since you only see the change(s) then.

JakeWharton commented:
Merged manually. Thanks!
